app/Dao/UserDao.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace App\Dao;
+
+use App\Contracts\Dao\UserDaoInterface;
+use App\Models\User;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+
+class UserDao implements UserDaoInterface
+{
+    /**
+     * Insert Data
+     *
+     * @param array $insertData
+     * @return void
+     */
+    public function insert(array $insertData): void
+    {
+        $insertData['role'] = '2';
+        User::create($insertData);
+    }
+    
+    /**
+     * Get Users By Id
+     *
+     * @param int $id
+     * @return User
+     */
+    public function getUserById(int $id): User
```

Comment:
```suggestion
     * @return User|null
     */
    public function getUserById(int $id): User|null
```
Return type can be either user or null when id doesn't match any record.